File: leetcode/1323-maximum-69-number/solution.ts
function maximum69Number(num: number): number {
  let numCopy = num;

  let i = 0;
  let lastSixPos = -1;

  while (numCopy > 0) {
    if (numCopy % 10 == 6) {
      lastSixPos = i;
    }

    numCopy = Math.floor(numCopy / 10);
    i++;
  }

  if (lastSixPos == -1) {
    return num;
  }

  return num + 3 * Math.pow(10, lastSixPos);
}

File: leetcode/1323-maximum-69-number/solution.cpp
int maximum69Number (int num) {
    int num_copy = num;

    int i = 0;
    int last_six = -1;

    while (num_copy) {
        if (num_copy % 10 == 6) {
            last_six = i;
        }

        num_copy /= 10;
        i++;
    }

    if (last_six == -1) {
        return num;
    }

    return num + 3 * pow(10, last_six);
}